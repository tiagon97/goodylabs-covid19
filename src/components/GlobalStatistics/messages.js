const messages = {
  header: 'Daily global statistics',
  cases: 'COVID-19 cases:',
  deaths: 'Total deaths:',
  recovered: 'Total recovered:',
};

export default messages;
