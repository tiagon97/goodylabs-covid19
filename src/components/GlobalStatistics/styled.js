import styled from 'styled-components';

import { Colors } from '../../constants';

export const Wrapper = styled.div`
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  background-color: #eee;
  text-align: center;
  padding: 40px 0;
  font-weight: 600;
  font-size: 2rem;
`;

const handleColorGenre = (color) => {
  switch (color) {
    case 'primary':
      return Colors.yellow;
    case 'secondary':
      return Colors.red;
    default:
      return Colors.green;
  }
};

export const StyledSpan = styled.span`
  color: ${({ color }) => handleColorGenre(color)};
`;

export const Tile = styled.div`
  display: flex;
  justify-content: center;
`;

export const Title = styled.span`
  margin-right: 8px;
`;

export const Data = styled.span``;
