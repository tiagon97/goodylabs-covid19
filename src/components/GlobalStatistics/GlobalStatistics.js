import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import format from '../../utils';

const Wrapper = styled.div`
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  background-color: #eee;
  text-align: center;
  padding: 40px 0;
  font-weight: 600;
  font-size: 2rem;
`;

const handleColorGenre = (color) => {
  switch (color) {
    case 'primary':
      return '#DAA520';
    case 'secondary':
      return 'red';
    default:
      return '#32CD32';
  }
};

const StyledSpan = styled.span`
  color: ${({ color }) => handleColorGenre(color)};
`;

const GlobalStatistics = ({ globalData }) => {
  return (
    <>
      <h2>Global statistics: </h2>
      <Wrapper>
        <p>
          COVID-19 cases:{' '}
          {globalData.TotalConfirmed ? (
            <StyledSpan color="primary">{format`${globalData.TotalConfirmed}`}</StyledSpan>
          ) : (
            'loading...'
          )}
        </p>
        <p>
          Total deaths:{' '}
          {globalData.TotalConfirmed ? (
            <StyledSpan color="secondary">{format`${globalData.TotalDeaths}`}</StyledSpan>
          ) : (
            'loading...'
          )}
        </p>
        <p>
          Total recovered:{' '}
          {globalData.TotalConfirmed ? <StyledSpan>{format`${globalData.TotalRecovered}`}</StyledSpan> : 'loading...'}
        </p>
      </Wrapper>
    </>
  );
};

GlobalStatistics.propTypes = {
  globalData: PropTypes.instanceOf(Object).isRequired,
};

export default GlobalStatistics;
