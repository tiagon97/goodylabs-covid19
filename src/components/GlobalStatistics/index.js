import React from 'react';
import PropTypes from 'prop-types';

import Heading from '../Heading';
import CircleSpinner from '../CircleSpinner';
import format from '../../utils';
import messages from './messages';
import { Wrapper, StyledSpan, Tile, Title, Data } from './styled';

const GlobalStatistics = ({ globalData }) => (
  <>
    <Heading>{messages.header}</Heading>
    <Wrapper>
      <Tile>
        <Title>{messages.cases}</Title>
        <Data>
          {globalData.TotalConfirmed ? (
            <StyledSpan color="primary">{format`${globalData.TotalConfirmed}`}</StyledSpan>
          ) : (
            <CircleSpinner />
          )}
        </Data>
      </Tile>
      <Tile>
        <Title>{messages.deaths}</Title>
        <Data>
          {globalData.TotalDeaths ? (
            <StyledSpan color="secondary">{format`${globalData.TotalDeaths}`}</StyledSpan>
          ) : (
            <CircleSpinner />
          )}
        </Data>
      </Tile>
      <Tile>
        <Data>
          <Title>{messages.recovered}</Title>
          {globalData.TotalRecovered ? (
            <StyledSpan>{format`${globalData.TotalRecovered}`}</StyledSpan>
          ) : (
            <CircleSpinner />
          )}
        </Data>
      </Tile>
    </Wrapper>
  </>
);

GlobalStatistics.propTypes = {
  globalData: PropTypes.objectOf(PropTypes.number).isRequired,
};

export default GlobalStatistics;
