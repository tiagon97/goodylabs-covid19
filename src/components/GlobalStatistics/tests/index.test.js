/**
 *
 * Tests for GlobalStatistics
 *
 */

import React from 'react';
import { render } from '@testing-library/react';

import GlobalStatistics from '../index';

const defaultProps = {
  globalData: {
    TotalConfirmed: 100,
    TotalDeaths: 10,
    TotalRecovered: 50,
  },
};

describe('<GlobalStatistics />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<GlobalStatistics {...defaultProps} />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<GlobalStatistics {...defaultProps} />);
    expect(firstChild).toMatchSnapshot();
  });
});
