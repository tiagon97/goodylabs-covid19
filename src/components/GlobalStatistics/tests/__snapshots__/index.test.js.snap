// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GlobalStatistics /> Should render and match the snapshot 1`] = `
<h1
  class="sc-AxjAm cYlDMx"
>
  Daily global statistics
</h1>
`;
